packages/client/src/declarations/index.ts

```diff
@@ -721,8 +736,37 @@ export async function getDeclarationsOfCurrentUser(): Promise<string> {
   const currentUserData = allUserData.find(
     (uData) => uData.userID === currentUserID
   )
-  const currentUserDeclarations: IDeclaration[] =
+  let currentUserDeclarations: IDeclaration[] =
     (currentUserData && currentUserData.declarations) || []
+
+  if (Roles.FIELD_AGENT.includes(currentUserRole) && currentUserData) {
+    currentUserDeclarations = currentUserData.declarations.filter((d) => {
+      if (d.downloadStatus === DOWNLOAD_STATUS.DOWNLOADED) {
+        const history = d.originalData?.history as unknown as IDynamicValues
+
+        const downloadedTime = (
+          history.filter((h: IDynamicValues) => {
+            return h.action === DOWNLOAD_STATUS.DOWNLOADED
+          }) as IDynamicValues[]
+        ).sort((fe, se) => {
+          return new Date(se.date).getTime() - new Date(fe.date).getTime()
+        })
+
+        return (
+          differenceInMinutes(new Date(), new Date(downloadedTime[0].date)) <
+          1440 // 24 hours, used munites for better accuracy
+        )
+      }
+      return true
+    })
+  }
+
+  // Storing the declarations again by excluding the 24 hours old downloaded declaration
+  if (currentUserData) {
+    currentUserData.declarations = currentUserDeclarations
+    await storage.setItem('USER_DATA', JSON.stringify(allUserData))
```

Comment:
@atiqzaman-dsi this would clear USER_DATA into an empty array for all users who are not FIELD_AGENT.  You should only do this inside the above conditional